heron/packing/src/java/org/apache/heron/packing/roundrobin/RoundRobinPacking.java

```diff
@@ -463,11 +463,15 @@ private ByteAmount getContainerDiskHint(Map<Integer, List<InstanceId>> allocatio
    * @return Container RAM requirement
    */
   private ByteAmount getContainerRamHint(Map<Integer, List<InstanceId>> allocation) {
+    ByteAmount defaultContainerRam = instanceRamDefault
```

Comment:
same as #3187, could it be helpful to refactor this logic into a function? since the logic is used in a few places.